scan/dependencies.go

```diff
@@ -147,6 +147,9 @@ func NewImageReference(imagePath string) (*image.Reference, error) {
 	if tagged, ok := ref.(reference.Tagged); ok {
 		result.Tag = tagged.Tag()
 	}
+	if digested, ok := ref.(reference.Digested); ok {
```

Comment:
Just curious, what is the change for? Is it to fix some related bug?

Reply:
This change is for dockerfiles like:

```Dockerfile
FROM busybox@sha256:f3cfc9d0dbf931d3db4685ec659b7ac68e2a578219da4aae65427886e649b06b
RUN echo "hello" > world.txt
```

As you see, `busybox` already is digested, there is no need to `docker inspect` it or call Manifest API to resolve the digest, we already resolve it while scanning dependencies.

That's why I added the following to return fast in `populateDigest()` method:

```go
if ref.Digest != ""  {
    return nil	
}
```